### PR TITLE
Add storageprofiles to RBAC

### PR DESCRIPTION
**Pull request by brybacki**
Signed-off-by: Bartosz Rybacki <brybacki@redhat.com>

**What this PR does / why we need it**:

Adds RBAC rules for storageprofiles

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

https://bugzilla.redhat.com/show_bug.cgi?id=1992961

**Special notes for your reviewer**:

**Release note**:

```release-note
Fix  access control for storageprofiles
```

### CODE REVIEW

**Comment by akalenyu** [comment]
> /test all

seeing this too, do we want to merge https://github.com/kubevirt/containerized-data-importer/pull/2019 maybe?
im not sure whats responsible to update the binary if at all

File: pkg/operator/resources/cluster/rbac.go [hunk 1]
```diff
@@ -84,6 +84,21 @@ func getAdminPolicyRules() []rbacv1.PolicyRule {
...
+			},
+			Resources: []string{
+				"storageprofiles",
+			},
+			Verbs: []string{
+				"get",
+				"list",
+				"watch",
+				"patch",
+				"update",
```
Comment (mhenriks) [inline pkg/operator/resources/cluster/rbac.go:99]: Don't see this tested anywhere

Comment (mhenriks) [reply pkg/operator/resources/cluster/rbac.go:99]: Sorry, I meant `update` specifically.  Basically should storageprofile (and cdiconfig) be in the admin/edit/view clusterroles?

Comment (brybacki) [reply pkg/operator/resources/cluster/rbac.go:99]: That is good question, I also wonder if it makes sens. 
Correct me if I am wrong, but reading here [1], I see that default admin, edit and view roles are intended to be granted within particular namespace. 

The test [var _ = Describe("Aggregated role in-action tests", func() {](https://github.com/kubevirt/containerized-data-importer/blob/d0a58c266bcccc187f2f6aed1c30b584a3eda1f9/tests/rbac_test.go#L26) passes thanks to the following clusterRole and Binding https://github.com/kubevirt/containerized-data-importer/blob/d0a58c266bcccc187f2f6aed1c30b584a3eda1f9/pkg/operator/resources/cluster/rbac.go#L187.



https://kubernetes.io/docs/reference/access-authn-authz/rbac/#default-roles-and-role-bindings

**Comment by mhenriks** [comment]
would you mind removing cdiconfig from the aggregate roles since they should not be there?

**Comment by brybacki** [comment]
Done

**Comment by brybacki** [comment]
/test pull-containerized-data-importer-e2e-k8s-1.21-ceph

Cleanup timeout, so not a big problem:
`                        Message: "Some content in the namespace has finalizers remaining: snapshot.storage.kubernetes.io/pvc-as-source-protection in 1 resource instances",
`

**Pull request merged**